core/src/main/java/apoc/export/util/MetaInformation.java

```diff
@@ -1,38 +1,65 @@
 
+import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import static apoc.gephi.GephiFormatUtils.getCaption;
+import static apoc.meta.tablesforlabels.PropertyTracker.typeMappings;
 import static java.util.Arrays.asList;
+import static org.neo4j.internal.helpers.collection.Iterables.stream;
 
 /**
  * @author mh
  * @since 19.01.14
  */
 public class MetaInformation {
+    
+    private static final Map<String, String> REVERSED_TYPE_MAP = MapUtils.invertMap(typeMappings);
+    
+    public static Map<String, Class> collectPropTypesForNodes(SubGraph graph, GraphDatabaseService db, ExportConfig config) {
+        final Map<String, Object> conf = new HashMap<>(config.getConfig());
+        conf.put("includeLabels", stream(graph.getAllLabelsInUse()).map(Label::name).collect(Collectors.toList()));
+        
+        return db.executeTransactionally("CALL apoc.meta.nodeTypeProperties($conf)", 
+                Map.of("conf", conf), getMapResultTransformer()); 
+    }
 
-    public static Map<String,Class> collectPropTypesForNodes(SubGraph graph) {
-        Map<String,Class> propTypes = new LinkedHashMap<>();
-        for (Node node : graph.getNodes()) {
-            updateKeyTypes(propTypes, node);
-        }
-        return propTypes;
+    public static Map<String, Class> collectPropTypesForRelationships(SubGraph graph, GraphDatabaseService db, ExportConfig config) {
+        final Map<String, Object> conf = new HashMap<>(config.getConfig());
+        conf.put("includeRels", stream(graph.getAllRelationshipTypesInUse()).map(RelationshipType::name).collect(Collectors.toList()));
+
+        return db.executeTransactionally("CALL apoc.meta.relTypeProperties($conf)", 
+                Map.of("conf", conf), getMapResultTransformer());
     }
-    public static Map<String,Class> collectPropTypesForRelationships(SubGraph graph) {
-        Map<String,Class> propTypes = new LinkedHashMap<>();
-        for (Relationship node : graph.getRelationships()) {
-            updateKeyTypes(propTypes, node);
-        }
-        return propTypes;
+
+    private static ResultTransformer<Map<String, Class>> getMapResultTransformer() {
+        return result -> result.stream()
+                .filter(map -> map.get("propertyName") != null)
+                .collect(Collectors.toMap(map -> (String) map.get("propertyName"),
+                        map -> {
+                            final String propertyTypes = ((List<String>) map.get("propertyTypes")).get(0);
+                            String className = REVERSED_TYPE_MAP.get(propertyTypes);
+                            try {
+                                return ClassUtils.getClass(className);
+                            } catch (ClassNotFoundException e) {
+                                throw new RuntimeException(e);
+                            }
```

Comment:
questo non l'ho capito proprio

Reply:
Qui il risultato delle procedure meta lo passo alla  `REVERSED_TYPE_MAP`, 
che è l'inverso della `typeMappings` usata per restituire i risultati di `meta.nodeTypeProperties` e `meta.relTypeProperties` in base alla Java Class. 
Quindi qui faccio l'inverso, dal risultato delle meta mi ricavo la Classe Java